code/modules/reagents/chemistry/machinery/reagentgrinder.dm

```diff
@@ -205,6 +206,26 @@
 		holdingitems[weapon] = TRUE
 		return FALSE
 
+/obj/machinery/reagentgrinder/proc/on_storage_dump(datum/source, datum/storage/storage, mob/user)
+	SIGNAL_HANDLER
+
+	for(var/obj/item/to_dump in storage.real_location)
+		if(holdingitems.len >= limit)
+			break
+
+		if(!to_dump.grind_results && !to_dump.juice_typepath)
+			continue
+
+		if(!storage.attempt_remove(to_dump, src, silent = TRUE))
+			continue
+
+		holdingitems[to_dump] = TRUE
+		to_dump.pixel_x = to_dump.base_pixel_x + rand(-5, 5)
+		to_dump.pixel_y = to_dump.base_pixel_y + rand(-5, 5)
```

Comment:
```suggestion
```
Not sure why you're randomizing pixel_x and pixel_y when they're not going to be visible